src/telemetry/metrics/labels.rs

```diff
@@ -79,7 +74,7 @@ impl RequestLabels {
     pub fn new(req: &ctx::http::Request) -> Self {
         let direction = Direction::new(req.server.proxy);
 
-        let outbound_labels = req.dst_labels().cloned();
+        let outbound_labels = DstLabels::new(req.labels());
```

Comment:
> Previously, dst label strings would be formatted eagerly at
> service-discovery-time. This change moves this string formatting into
> the data path(!)

Why not do the string formatting at metrics format time instead? The `RequestLabels` could have an `Arc<IndexMap<...>>` and format it on the control thread when the metrics are served. This would hurt the metrics scrape performance a bit, but it seems to me that minimizing work in the data path is more important?

If we're concerned about doing the same work multiple times, we could also use interior mutability to have `DstLabels` memoize the string once we've formatted it once...this would admittedly add a bit of complexity, but it could at least be contained inside the label type.

Reply:
We'll get there. Currently, `DstLabels` needs to impl `Hash`, but `IndexMap` doesn't impl `Hash`.  I'm working on breaking these requirements so that we can format the strings at Fmt-time.